src/native/fs/fs_napi.cpp

```diff
@@ -285,6 +296,92 @@ struct Stat : public FSWorker
     }
 };
 
+/**
+ * IsEncrypted is an spectrumscale op
+ */
+struct IsEncrypted : public FSWorker
+{
+    std::string _path;
+    bool _is_encrypt_res;
+
+    IsEncrypted(const Napi::CallbackInfo& info)
+        : FSWorker(info)
+    {
+        _path = info[1].As<Napi::String>();
+        Begin(XSTR() << "IsEncrypted " << DVAL(_path));
+    }
+    virtual void Work()
+    {
+        int fd, rc;
+        char *bufP = NULL;
+        int bufLen = 256;
+        int nameLen;
+        gpfsFcntlHeader_t *hdrP;
+        gpfsGetSetXAttr_t *getAttrP;
+        char xnambuf[256];
+        _is_encrypt_res = false;
+
+        strcpy(xnambuf,XATTR_APPID_TO_NAMESPACE(GPFS_ATTR_INTERNAL_APPL_ID));
+        strcat(xnambuf,".");
+        strcat(xnambuf,GPFS_DOT_ENCRYPTION_EA);
+
+        nameLen = strlen(xnambuf);
+
+        fd = open(_path.c_str(), O_RDONLY);
```

Comment:
This is probably a case where we would prefer to operate on an open file instead of a path, which means that instead of making this function an FSWorker it will be added as a method to FileWrap.

Reply:
@guymguym If we will add  is_encrypted to FileWrap doesn't it always to be in that class? will I just return false when it's not GPFS drvie or something like this, also adding this function as part of the gpfs submodule that you asked for also won't be needed. Do I get you right here?

Reply:
added FileWrap is_encrypted functionality